test/boost/mutation_test.cc

```diff
@@ -687,14 +687,11 @@ SEASTAR_TEST_CASE(test_cell_ordering) {
     auto expiry_2 = now + ttl_2;
 
     auto assert_order = [] (atomic_cell_view first, atomic_cell_view second) {
-        if (compare_atomic_cell_for_merge(first, second) >= 0) {
-            testlog.trace("Expected {} < {}", first, second);
-            abort();
-        }
-        if (compare_atomic_cell_for_merge(second, first) <= 0) {
-            testlog.trace("Expected {} < {}", second, first);
-            abort();
-        }
+        testlog.trace("Expected {} < {}", first, second);
```

Comment:
The trace logs are needed since we can't use `BOOST_REQUIRE_LT(compare_atomic_cell_for_merge(first, second), 0)` because boost doesn't know how to print `std::strong_ordering` and the boost printout on failure is too cryptic for `BOOST_REQUIRE(compare_atomic_cell_for_merge(first, second) < 0)`, making the trace logging useful when trying to figure out what exactly happened (the comparison can fail 2 ways: when comparing left, right and when comparing right, left).